lib/kubernetes-deploy/options_helper.rb

```diff
@@ -6,22 +6,35 @@ class OptionsError < StandardError; end
 
     STDIN_TEMP_FILE = "from_stdin.yml.erb"
     class << self
-      def with_validated_template_dir(template_dir)
-        if template_dir == '-'
+      def with_validated_template_dirs(template_dirs)
+        if template_dirs.select { |dir| dir == "-" }.length > 2
+          raise OptionsError, "Cannot specify stdin as a template directory more than once"
+        end
+
+        dirs = []
+        if template_dirs.empty?
+          dirs << default_template_dir
+        else
+          template_dirs.each do |template_dir|
+            next if template_dir == '-'
+            dirs << template_dir
+          end
+        end
+
+        if template_dirs.include?("-")
           Dir.mktmpdir("kubernetes-deploy") do |dir|
             template_dir_from_stdin(temp_dir: dir)
-            yield dir
+            dirs << dir
+            yield dirs
```

Comment:
This control flow feels a bit clumsy, but it accomplishes the goal of wrapping `mktmpdir` inside the execution of the deploy task, ensuring it will be cleaned up on program exit